src/precice/impl/DataContext.hpp

```diff
@@ -58,6 +58,11 @@ class DataContext {
    */
   MeshID getMeshID() const;
 
+  /**
+   * @brief Get the Mesh
+   */
+  mesh::PtrMesh getMesh() const;
```

Comment:
I'm not a big fan of exposing the mesh via the data context, because one might do bad things with the mesh then. As far as I see, you are just exposing the mesh for 1) getting the dimension (`mesh.getDimensions()`), 2) getting the number of vertices (`mesh.vertices().size()`), and 3) checking the vertex ids (`isValidVertexID(vid)`). Why not provide dedicated functions for this for better encapsulation?

Reply:
The solution to this is `const mesh::Mesh& getMesh() const;`.

I am not a big fan of unnecessary wrappers. They make refactoring unnecessary painful.

Reply:
I don't see these wrappers as unnecessary. In the past introducing such wrappers helped me a lot when refactoring the code base. From my perspective one main problem in our design is that often there are long chains of calls that make it really hard to understand in which context, for example, `Mesh::vertices()` is used. `vertices().size()` has around 200 hits in our code base. Introducing a wrapper like `Mesh::verticesSize()` allows me to ignore these 200 hits and concentrate on the rest.